Windows - Configuration error when building from source - dlpack symbolic link

### Description

When trying to build from source on Windows 11 with CUDA 11.8 the build failed as the symbolic link to the third party dlpack library folder did not work as intended. 

The relevant error output I got was:
```
-------- Configuring Module: dlpack --------
      a.cpp
      C:\Users\woute\Projects\cupy\cupy/_core/include\cupy/_dlpack/dlpack.h(1): error C2059: syntax error: '.'
      C:\Users\woute\AppData\Local\Temp\tmpxbm1x8ax\a.cpp(2): error C2143: syntax error: missing ';' before '{'
      C:\Users\woute\AppData\Local\Temp\tmpxbm1x8ax\a.cpp(2): error C2447: '{': missing function header (old-style formal list?)
      command 'C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\MSVC\\14.34.31933\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
      a.cpp
      Generating code
      Finished generating code
```

### To Reproduce

```py
py -m pip install .
```


### Installation

Built from GitHub source

### Additional Information

I fixed it by changing https://github.com/cupy/cupy/blob/cf9b4a517725b803fba4828ed0b1c93a231cc5da/cupy/_core/include/cupy/_dlpack/dlpack.h#L1

to 
```c
#include "../../../../../third_party/dlpack/include/dlpack/dlpack.h"
```

After changing this line the build started but crashed elsewhere. But that is for another issue :)

--- a/docs/source/contribution.rst
+++ b/docs/source/contribution.rst
@@ -412,3 +412,14 @@ For example, if you only run your CuPy build with NVIDIA P100 and V100, you can 
   $ export CUPY_NVCC_GENERATE_CODE=arch=compute_60,code=sm_60;arch=compute_70,code=sm_70
 
 See :doc:`reference/environment` for the description.
+
+Development on Microsoft Windows
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+CuPy uses symbolic links in the source tree.
+If you are developing on Windows (outside of WSL), you will need additional setup before cloning the CuPy repository.
+
+* Run ``git config --global core.symlinks true`` to enable symbolic link support in Git.
+* `Activate Developer Mode <https://learn.microsoft.com/en-us/windows/apps/get-started/enable-your-device-for-development#activate-developer-mode>`_ to allow Git using symbolic link without Administrator privilege.
+
+Once configured, you can clone the repository by ``git clone --recursive https://github.com/cupy/cupy.git``.

